src/node/ccf_acme_client.h

```diff
@@ -31,21 +39,133 @@ namespace ccf
     }
   }
 
+  class ACMERpcEndpoints : public CommonEndpointRegistry
+  {
+  public:
+    ACMERpcEndpoints(
+      NetworkState& network, ccfapp::AbstractNodeContext& context) :
+      CommonEndpointRegistry(get_actor_prefix(ActorsType::well_known), context)
+    {
+      auto handler = [this](auto& ctx) {
+        http_status response_status = HTTP_STATUS_INTERNAL_SERVER_ERROR;
+        std::string response_body;
+
+        try
+        {
+          const auto& path_params = ctx.rpc_ctx->get_request_path_params();
+          const auto url_token_it = path_params.find("token");
+
+          if (url_token_it == path_params.end())
+          {
+            ctx.rpc_ctx->set_response_status(HTTP_STATUS_NOT_FOUND);
+            ctx.rpc_ctx->set_response_body("no token in URL");
+          }
+
+          std::string token = url_token_it->second;
+          LOG_DEBUG_FMT("ACME: challenge request for token '{}'", token);
+
+          auto tit = prepared_responses.find(token);
+          if (tit == prepared_responses.end())
+          {
+            auto prit = prepared_responses.find("");
+            if (prit != prepared_responses.end())
+            {
+              response_status = HTTP_STATUS_OK;
+              response_body = token + "." + prit->second;
+            }
+            else
+            {
+              response_status = HTTP_STATUS_NOT_FOUND;
+              response_body = fmt::format(
+                "Challenge response for token '{}' not found", token);
+            }
+          }
+          else
+          {
+            response_status = HTTP_STATUS_OK;
+            response_body = token + "." + tit->second;
+          }
+        }
+        catch (const std::exception& ex)
+        {
+          response_status = HTTP_STATUS_INTERNAL_SERVER_ERROR;
+          response_body = ex.what();
+        }
+
+        ctx.rpc_ctx->set_response_status(response_status);
+        ctx.rpc_ctx->set_response_body(std::move(response_body));
+      };
+
+      make_endpoint(
+        "/acme-challenge/{token}", HTTP_GET, handler, no_auth_required)
```

Comment:
Is the endpoint supposed to return a JSON object? If so, the `json_adapter` could be used here.

Reply:
No, it's supposed to return just a string (`response_body = token + "." + tit->second;`)